hybrid/flow.py

```diff
@@ -587,13 +588,107 @@ def next(self, states):
         return states[min_idx]
 
 
-class Loop(Runnable):
+class TrackMin(Runnable, traits.SISO):
+    """Tracks (and keeps record of) the "best" (or minimal) :class:`State` seen,
+    according to a metric defined with a `key` function.
```

Comment:
If the metric is defined by a key, is the "best" always (or minimal)?

Tracks and records the best :class:`State` according to a metric defined with a `key` function; typically this is the minimal :class:`State`